vm-setup/roles/v1a2_integration_test/tasks/deprovision.yml

```diff
@@ -0,0 +1,39 @@
+---
+  - name: Deprovision machines
+    k8s:
+      state: absent
+      api_version: "{{ API_VERSION }}"
+      kind: Machine
+      namespace: "{{ NAMESPACE }}"
+      name: "{{ item }}"
+    loop:
+       - "{{ CLUSTER_NAME }}-controlplane-0"
+       - "{{ CLUSTER_NAME }}-md-0"
```

Comment:
This should not be the name. Since it comes from a machine set, it will have a part of random string. you should rather delete the machine deployment that has this name